packages/openapi-generator/src/parser/document.ts

```diff
@@ -41,13 +42,27 @@ export function parseAllOperations(
   refs: $Refs
 ): OpenApiOperation[] {
   return Object.entries(document.paths).reduce(
-    (allOperations, [pattern, pathDefinition]) => [
+    (allOperations, [path, pathDefinition]) => [
       ...allOperations,
       ...methods
         .filter(method => pathDefinition?.[method])
         // Undefined path definitions have been filtered out in the line before
-        .map(method => parseOperation(pattern, pathDefinition!, method, refs))
+        .map(method => parseOperation(path, pathDefinition!, method, refs))
     ],
     []
   );
 }
+
+/**
+ * Collect all the tags used by given operations.
+ * @param operations The given operations.
+ * @returns An array that holds the unique tags.
+ */
+export function collectTags(operations: OpenApiOperation[]): string[] {
+  const tags: string[] = [];
+  operations.forEach(o => {
+    const tagsOfOperation = o.tags?.length ? o.tags : ['default'];
+    tagsOfOperation.forEach(tag => tags.push(tag));
```

Comment:
you could use:     `['c'].push(...['a','b'])` and I guess the most JS like way would be to use the `reduce` method to collect all tags in one list @marikaner what do you say as the JS native:  :-).